src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/MidEventCallback.java

```diff
@@ -0,0 +1,96 @@
+package uk.gov.hmcts.ccd.domain.service.createevent;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.definition.CachedCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.definition.CaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.definition.UIDefinitionRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseEvent;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseType;
+import uk.gov.hmcts.ccd.domain.model.definition.WizardPage;
+import uk.gov.hmcts.ccd.domain.model.std.Event;
+import uk.gov.hmcts.ccd.domain.service.common.CaseService;
+import uk.gov.hmcts.ccd.domain.service.common.EventTriggerService;
+import uk.gov.hmcts.ccd.domain.service.stdapi.CallbackInvoker;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+@Service
+public class MidEventCallback {
+
+    private CallbackInvoker callbackInvoker;
+    private UIDefinitionRepository uiDefinitionRepository;
+    private EventTriggerService eventTriggerService;
+    private CaseDefinitionRepository caseDefinitionRepository;
+    private CaseService caseService;
+
+    @Autowired
+    public MidEventCallback(CallbackInvoker callbackInvoker, UIDefinitionRepository uiDefinitionRepository,
+                            EventTriggerService eventTriggerService,
+                            @Qualifier(CachedCaseDefinitionRepository.QUALIFIER) CaseDefinitionRepository caseDefinitionRepository,
+                            CaseService caseService) {
+        this.callbackInvoker = callbackInvoker;
+        this.uiDefinitionRepository = uiDefinitionRepository;
+        this.eventTriggerService = eventTriggerService;
+        this.caseDefinitionRepository = caseDefinitionRepository;
+        this.caseService = caseService;
+    }
+
+    public JsonNode invoke(String jurisdictionId, String caseTypeId, Event event,
+                                        Map<String, JsonNode> data,
```

Comment:
3 parameters in one line then one in a line.  Should be trivial to make the code more readable